include/fmt/chrono.h

```diff
@@ -1091,21 +1105,51 @@ template <typename OutputIt, typename Char> class tm_writer {
     out_ = copy_str<Char>(std::begin(buf) + offset, std::end(buf), out_);
   }
 
-  void on_utc_offset() { format_localized('z'); }
+  void on_utc_offset() {
+#if defined(_WIN32)
+    _tzset();
+    long offset = 0;
+    _get_timezone(&offset);
+    if (tm_.tm_isdst) {
+      long dstbias = 0;
+      _get_dstbias(&dstbias);
+      offset += dstbias;
+    }
+    write_utc_offset(-offset);
+#elif defined(__APPLE__) || defined(__GLIBC__) || defined(__FreeBSD__) || \
+    defined(__NetBSD__) || defined(__OpenBSD__) || defined(__DragonFly__)
+    write_utc_offset(tm_.tm_gmtoff);
+#elif defined(sun) || defined(__sun)
+    tzset();
+    write_utc_offset(
+        static_cast<long>(tm_.tm_isdst == 0 ? -timezone : -altzone));
+#else
+    // For exotic platforms.
+    format_localized('z');
+#endif
+  }
   void on_tz_name() { format_localized('Z'); }
 
   void on_year(numeric_system ns) {
-    if (ns != numeric_system::standard) return format_localized('Y', 'E');
+    if (!is_classic_ && ns != numeric_system::standard)
+      return format_localized('Y', 'E');
     write_year(tm_year());
```

Comment:
Let's swap the branches to get rid of negation:

```c++
    if (is_classic_ || ns == numeric_system::standard)
      return write_year(tm_year());
    format_localized('Y', 'E');
```

and the same below.

Reply:
Done

Reply:
``tm_gmtoff`` and SFINAE - Done.
Solaris specific code is removed.
Windows specific code I suggest to include in fmt, because Windows is more popular.

Reply:
I was wrong.
Inside ``ucrt``, the internal function __tzset () is used, which is called only once. When using _tzset (), performance is 10 times slower.

I've added a workaround, but I'm not sure if I need to switch to custom code.

Perf with workaround:

```
2021-11-22T00:37:43+05:00
Running fmt_test_old
Run on (2 X 2494 MHz CPU s)
CPU Caches:
  L1 Data 32 KiB (x2)
  L1 Instruction 32 KiB (x2)
  L2 Unified 4096 KiB (x2)
  L3 Unified 16384 KiB (x1)
------------------------------------------------------------------
Benchmark                        Time             CPU   Iterations
------------------------------------------------------------------
FMTFormatter_z                1384 ns         1360 ns       448000
FMTFormatterCompile_z         1321 ns         1318 ns       497778
FMTFormatter_z_strftime        312 ns          314 ns      2240000
FMTFormatter_z_time_put       1217 ns         1221 ns       640000
FMTFormatter_Y                1385 ns         1381 ns       497778
FMTFormatterCompile_Y         1300 ns         1311 ns       560000
FMTFormatter_Y_strftime        301 ns          300 ns      2240000
FMTFormatter_Y_time_put       1214 ns         1221 ns       640000
2021-11-22T00:37:50+05:00
Running fmt_test_new
Run on (2 X 2494 MHz CPU s)
CPU Caches:
  L1 Data 32 KiB (x2)
  L1 Instruction 32 KiB (x2)
  L2 Unified 4096 KiB (x2)
  L3 Unified 16384 KiB (x1)
------------------------------------------------------------------
Benchmark                        Time             CPU   Iterations
------------------------------------------------------------------
FMTFormatter_z                 123 ns          123 ns      5600000
FMTFormatterCompile_z         71.1 ns         69.8 ns      8960000
FMTFormatter_z_strftime        312 ns          314 ns      2240000
FMTFormatter_z_time_put       1214 ns         1221 ns       640000
FMTFormatter_Y                 121 ns          120 ns      5600000
FMTFormatterCompile_Y         67.9 ns         68.4 ns     11200000
FMTFormatter_Y_strftime        303 ns          305 ns      2357895
FMTFormatter_Y_time_put       1203 ns         1200 ns       560000
```